syntax = "proto2";
package tutorial;

//common response
message rsponse_result {
	required string msg  = 2;
	optional string data = 3;
	required int32 code  = 1;
}

//get sms code
message mobile_request {
	required string mobile  = 1;
}

message login_request {
	required string mobile = 1;
    required string code   = 2;
}

message recharge_request {
	required string mobile = 1;
    required int32 amount = 2;
}


message account_balance_request {
	required string mobile = 1;
}

// sub struct 
message account_balance_response {
	required tutorial.rsponse_result ret = 1;
	required int32 balance = 2;
}

// TODO: why too less
message list_account_records_request {
	required string mobile = 1;
}

message list_account_records_response {
	required tutorial.rsponse_result ret = 1;
	repeated tutorial.list_account_records_response.account_record records = 2;

	message account_record {
		required int32 type = 1;
		required int32 limit = 2;
		required uint64 timestamp = 3;
	}
}

message list_travel_records_request {
	required string mobile = 1;
}

//嵌套方式,生成typedef
message list_travel_records_rsponse {
	required tutorial.rsponse_result ret   = 1;
	required double mileage   = 2;
	required double discharge = 3;
	required double calorie   = 4;
	repeated tutorial.list_travel_records_rsponse.travel_record records = 5;

	message travel_record {
		required uint64 stm      = 1;
		required uint32 duration = 2;
		required uint32 amount   = 3;
	}
}

message lock_request {
	required string mobile = 1;
	required string bike_code = 2;
}


message list_account_record_request {
	required string mobile = 1;
}


message unlock_request {
	required string mobile = 1;
	required string bike_code = 2;
}
